File: 883-car-fleet/car-fleet.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        int n = position.size();
        vector<pair<int,double>> pos;
        int fleetCount = 0;

        for(int i=0; i<n; i++){
            double time = (double)(target - position[i]) / speed[i];
            pos.push_back({position[i], time});
        }
        sort(pos.rbegin(), pos.rend());

        double lastTime = 0;

        for(int i=0; i<n; i++){
            if(pos[i].second > lastTime){
                fleetCount++;
                lastTime = pos[i].second;
            }
        }
        return fleetCount;
    }
};

File: 883-car-fleet/car-fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        pair = [[p, s] for p,s in zip(position, speed)]
        lastCar = 0
        fleets = 0

        for [p,s] in sorted(pair, reverse=True):
            time = (target - p)/s
            if(time > lastCar):
                fleets += 1
                lastCar = time

        return fleets